mtags-shared/src/main/scala/scala/meta/internal/metals/ClasspathSearch.scala

```diff
@@ -57,21 +57,27 @@ class ClasspathSearch(
   private def search(
       query: WorkspaceSymbolQuery,
       shouldVisitPackage: String => Boolean,
-      isCancelled: () => Boolean
+      isCancelled: () => Boolean,
+      isRetry: Boolean = false
   ): Iterator[Classfile] = {
-    for {
-      pkg <- packages.iterator
-      if pkg.packages.exists(shouldVisitPackage)
-      if !isCancelled()
-      if query.matches(pkg.bloom)
-      classfile <- pkg.members
-      if classfile.isClassfile
-      isMatch = {
-        if (query.isExact) Fuzzy.isExactMatch(query.query, classfile.filename)
-        else query.matches(classfile.fullname)
-      }
-      if isMatch
-    } yield classfile
+    val result =
+      for {
+        pkg <- packages.iterator
+        if pkg.packages.exists(shouldVisitPackage)
+        if !isCancelled()
+        if query.matches(pkg.bloom)
+        classfile <- pkg.members
+        if classfile.isClassfile
+        isMatch = {
+          if (query.isExact && !isRetry)
```

Comment:
This seems to drop fuzzy match entirely.

Should we have a `Prefix` search type instead of reusing Exact? We could do the prefix if we have query.length < 3 ? So that the current behaviour stays but we show things better if we have 1 or 2 characters?

Reply:
I don't drop fuzzy match altogether for workspace symbols still fuzzy match is used, only for classpath I used prefix. Also it fallback to fuzzy if using prefix nothing was found (since we don't want empty completions). But I can use the same logic for both classpath and workspace of using prefix for < 3 chars and using fuzzy as fallback.

Reply:
I think overall fuzzy makes more sense since it's easy to forget exact name of a class. But with 2 or 1 character it's useless, so I would do prefix only then.

Of course we could gather some opinions on discord or scala contributors.